tests/benchmarks/test_join.py

```diff
@@ -0,0 +1,58 @@
+import dask.dataframe as dd
+import pytest
+
+from ..utils_test import cluster_memory, timeseries_of_size
+
+mem_mult = [
+    0.1,
+    pytest.param(
+        1,
+        marks=pytest.mark.skip(reason="Does not finish"),
+    ),
+    pytest.param(
+        10,
+        marks=pytest.mark.skip(reason="Does not finish"),
+    ),
+]
+
+
+@pytest.mark.parametrize("mem_mult", mem_mult)  # [0.1, 1, 10]
+def test_join_big(small_client, mem_mult):
+    memory = cluster_memory(small_client)  # 76.66 GiB
+
+    df1_big = timeseries_of_size(
+        memory * mem_mult,
+    )
+    df1_big["x2"] = df1_big["x"] * 1e9
+    df1_big = df1_big.astype({"x2": "int"})
+
+    df2_big = timeseries_of_size(
+        memory * mem_mult,
+    )
+
+    df2_big["x2"] = df2_big["x"] * 1e9
+    df2_big = df2_big.astype({"x2": "int"})
+
+    dd.merge(df1_big, df2_big, on="x2", how="inner").compute()
+
+
+@pytest.mark.parametrize("mem_mult", mem_mult)  # [0.1, 1, 10]
+def test_join_big_small(small_client, mem_mult):
+    memory = cluster_memory(small_client)  # 76.66 GiB
+
+    df_big = timeseries_of_size(
+        memory * mem_mult,
+    )
+
+    df_big["x2"] = df_big["x"] * 1e9
+    df_big = df_big.astype({"x2": "int"})
+
+    df_small = timeseries_of_size(
+        memory * 0.01,
+    )
+
+    df_small["x2"] = df_small["x"] * 1e9
+    df_small = df_small.astype({"x2": "int"})
+    df_small = df_small.repartition(npartitions=1)
```

Comment:
@jrbourbeau Adding this seems to add a lot on the time of the call. 
With repartition, I'm seeing a time of ~345 s see https://github.com/coiled/coiled-runtime/actions/runs/3154172388/jobs/5131528746#step:5:116 while if I don't have repartition I see ~67s https://github.com/coiled/coiled-runtime/actions/runs/3153958846/jobs/5131025373#step:5:109

Not sure what we want to do here.

Reply:
I think the best chance here will be to make df_small a pandas dataframe to see the behavior we expect instead of repartition to 1 partition. Currently, the small dataframe is 1% fo the cluster which would be around 700 MB Maybe this is too big for 1 partition. Need to explore more how this behaves.